### PR TITLE
Decide changing sequence dependencies on MX nodes according to resulting relation

### PR DESCRIPTION
**Pull request by onurctirtir**
When executing alter_table / undistribute_table udf's, we should not try
to change sequence dependencies on MX workers if new table wouldn't
require syncing metadata.
Previously, we were checking that for input table. But in some cases, the
fact that input table requires syncing metadata doesn't imply the same
for resulting table (e.g when undistributing a Citus table).
Even more, doing that was giving an unexpected error when undistributing
a Citus table so this commit actually fixes that.

Fixes https://github.com/citusdata/citus/issues/4710

### CODE REVIEW

**Comment by onurctirtir** [comment]
We didn't have the same problem in `alter_table` udf's but might worth to add such tests in this pr since we don't have

File: src/backend/distributed/commands/alter_table.c [hunk 1]
```diff
@@ -1178,7 +1186,7 @@ ReplaceTable(Oid sourceId, Oid targetId, List *justBeforeDropCommands,
 	{
 		changeDependencyFor(RelationRelationId, sequenceOid,
 							RelationRelationId, sourceId, targetId);
-		if (ShouldSyncTableMetadata(sourceId))
+		if (changeSequenceDependenciesOnWorkers && ShouldSyncTableMetadata(sourceId))
```
Comment (onderkalaci) [inline src/backend/distributed/commands/alter_table.c:1189]: Can we just make this `ShouldSyncTableMetadata (targetId)`?

Comment (onurctirtir) [reply src/backend/distributed/commands/alter_table.c:1189]: We should know for which table operation we are replacing the table, so I think the `tableId` wouldn't be suffficient itself

Comment (onurctirtir) [reply src/backend/distributed/commands/alter_table.c:1189]: Sorry, I misunderstood. This makes sense

**Review by onderkalaci — APPROVED** [review]
Can you please make sure that the PR description reflects the latest implementation?

**Pull request merged**